### PR TITLE
Fix RDO bug that was causing speed levels 1 and 2 to not work

### PR DESCRIPTION
**Pull request by fbossen**
Add assert statement to check that nearest MV is indeed the same
as the MV passed as argument to encode_block_b()

### CODE REVIEW

File: src/encoder.rs [hunk 1]
```diff
@@ -1770,7 +1778,9 @@ fn encode_partition_topdown(seq: &Sequence, fi: &FrameInvariants, fs: &mut Frame
 
     match partition {
         PartitionType::PARTITION_NONE => {
-            let part_decision = if !rdo_output.part_modes.is_empty() {
+            // FIXME: For now always redo RDO since prior RDO may have relied on inaccurate
+            // neighbours
```
Comment (ycho) [inline src/encoder.rs]: I am trying to understand which case redoing rdo_mode_decision()  is required.
I assume you have tested locally this patch stops the corruption.

Comment (fbossen) [reply src/encoder.rs]: The case is where you have a first split. Then RDO is done on 4 smaller blocks. The best parameters for the last 3 smaller blocks assume that the previous smaller blocks were not split. But these blocks may actually end up being split. In such case the NEARESTMV may not be the same anymore, which causes the issue.

Comment (ycho) [reply src/encoder.rs]: I think, in such case, even intra mode(s) of the 1st smaller block can change if it has split?

File: src/encoder.rs [hunk 2]
```diff
@@ -1770,7 +1778,9 @@ fn encode_partition_topdown(seq: &Sequence, fi: &FrameInvariants, fs: &mut Frame
...
+            let part_decision = if !rdo_output.part_modes.is_empty() && false {
```
Comment (ycho) [inline src/encoder.rs]: So, I am thinking that good chance of replacing 'false' here with sth better would be checking whether rdo_output.part_modes[0].bsize (does not exist currently) ==  current bsize.

**Pull request merged**